### PR TITLE
Add travis.yml files for each sub-package

### PR DESCRIPTION
**Pull request by alorenzen**
Add `.travis.yml` files for each sub-package, so that we can control the dart tasks for each sub-package without needing to write a custom script.

This also uses the `before_script` which is added in https://github.com/dart-lang/mono_repo/pull/38 in order to run our build before running the tests.

### CODE REVIEW

**Comment by alorenzen** [comment]
Let me know what you think about this change, and if you like it I can merge it internally.

File: _tests/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+language: dart
+sudo: false
+dart:
+  - dev
+
+dart_task:
+  - dartfmt
+  - dartanalyzer: --fatal-infos --fatal-warnings .
+  - test: -p vm -x codegen
+  - test: --precompiled=build --platform=chrome --tags=codegen --exclude-tags=known_pub_serve_failure -j1 test/common
```
Comment (kevmoo) [inline _tests/.travis.yml]: Duplicate line?

File: mono_repo.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Created by mono_repo
```
Comment (kevmoo) [inline mono_repo.yaml]: Does the auto-detect logic not work here? Ideally, this file shouldn't be needed...

**Comment by matanlurey** [comment]
Please do not approve PRs on GH.

We do not merge here.

File: _tests/.travis.yml (hunk 1 above)
Comment (alorenzen) [reply _tests/.travis.yml]: Yup. Good catch.

File: mono_repo.yaml (hunk 2 above)
Comment (alorenzen) [reply mono_repo.yaml]: Ah, ok. Wasn't sure what the best practice was for mono_repo. If we should check this in or not...
Guess the minor optimization of having this info cached is not worth the potential for stale data.

**Comment by alorenzen** [comment]
Quite a bit of build failures on travis. Not sure if they are related to my changes or not, so going to wait on merging this internally until folks are back from holidays and we can dig in.

**Comment by matanlurey** [comment]
Still want to do this? @alorenzen

**Comment by alorenzen** [comment]
Yes, I still intend to land this. I have been waiting for a slowdown in my `angular_ast` work to get back to this and rebase with recent changes from `build`. I think that will happen today.

**Comment by alorenzen** [comment]
I haven't yet re-run `mono_repo` after my latest merge, if that's what you're asking...

**Comment by kevmoo** [comment]
Ah. K. Are you going to?

**Comment by alorenzen** [comment]
I'm going to try to get the tests passing first.

**Comment by kevmoo** [comment]
Good idea...

**Comment by alorenzen** [comment]
Ok, this PR is now blocked on https://github.com/dart-lang/mono_repo/issues/43

`tool/travis.sh` is being treated as a binary file instead of a text file.

File: tool/travis.sh [hunk 3]
```diff
@@ -1,15 +1,109 @@
...
+  ;;
+test) echo
+  echo -e "\033[1mTASK: test\033[22m"
+  pub run test
+  ;;
+test_1) echo
+  echo -e "\033[1mTASK: test_1\033[22m"
+  pub run test -p vm -x codegen
+  ;;
+test_10) echo
```
Comment (kevmoo) [inline tool/travis.sh]: Ugh - https://github.com/dart-lang/mono_repo/issues/47

**Comment by kevmoo** [comment]
That's better!

File: tool/travis.sh [hunk 4]
```diff
@@ -1,15 +1,109 @@
...
+  tool/travis.sh
+  ;;
+angular_test) echo
+  echo -e "\033[1mPKG: angular_test\033[22m"
+  echo -e "  Running `tool/travis.sh`"
+  tool/travis.sh
+  ;;
+angular_router) echo
+  echo -e "\033[1mPKG: angular_router\033[22m"
+  echo -e "  Running `tool/travis.sh`"
```
Comment (kevmoo) [inline tool/travis.sh]: the script in the `ticks` is evaluated – I'm a moron

https://github.com/dart-lang/mono_repo/pull/49

**Pull request closed without merging**